File: articles/machine-learning-as-algorithmic-inference/go/threshold_review.go
package main

import "fmt"

func main() {
    tp, fp, tn, fn := 80.0, 25.0, 140.0, 35.0
    total := tp + fp + tn + fn
    accuracy := (tp + tn) / total
    precision := tp / (tp + fp)
    recall := tp / (tp + fn)
    fmt.Printf("accuracy=%.6f\n", accuracy)
    fmt.Printf("precision=%.6f\n", precision)
    fmt.Printf("recall=%.6f\n", recall)
}

File: articles/machine-learning-as-algorithmic-inference/haskell/ThresholdReview.hs
rate :: Double -> Double -> Double -> Double -> (Double, Double, Double)
rate tp fp tn fn = (accuracy, precision, recall)
  where
    total = max 1 (tp + fp + tn + fn)
    accuracy = (tp + tn) / total
    precision = tp / max 1 (tp + fp)
    recall = tp / max 1 (tp + fn)

main :: IO ()
main = do
  let (accuracy, precision, recallValue) = rate 80 25 140 35
  putStrLn ("accuracy=" ++ show accuracy)
  putStrLn ("precision=" ++ show precision)
  putStrLn ("recall=" ++ show recallValue)
